upgrade/kind.go

```diff
@@ -192,28 +192,19 @@ var getRepoKind = stepv2.Func11E("Get Repo Kind", func(ctx context.Context, repo
 	}
 
 	out.UpstreamProviderOrg = stepv2.Func11("Get UpstreamOrg", func(ctx context.Context, path string) string {
-		resourceFile := filepath.Join(path, "provider", "resources.go")
-		_, found := stepv2.Stat(ctx, resourceFile)
+		//resourceFile := filepath.Join(path, "provider", "resources.go")
+		//_, found := stepv2.Stat(ctx, resourceFile)
 		var upstreamOrg string
-		if found {
-			stepv2.SetLabel(ctx, "From resources.go")
-			resourceData := stepv2.ReadFile(ctx, resourceFile)
-			resLines := strings.Split(resourceData, "\n")
-
-			for _, line := range resLines {
-				if strings.Contains(line, "GitHubOrg") {
-					lineSlice := strings.Split(line, "\"")
-					upstreamOrg = lineSlice[1]
-					break
-				}
-			}
-		}
-		if upstreamOrg == "" {
-			// fall back to attempting to read from go.mod
-			stepv2.SetLabel(ctx, "Fall back to reading from go.mod")
-			tok := strings.Split(modPathWithoutVersion(upstream.Mod.Path), "/")
-			upstreamOrg = tok[len(tok)-2]
+		upstreamOrg = GetContext(ctx).UpstreamProviderOrg
+		if upstreamOrg != "" {
+			stepv2.SetLabel(ctx, "From Upgrade Config")
+			return upstreamOrg
 		}
+		// fall back to attempting to read from go.mod
+		stepv2.SetLabel(ctx, "Fall back to reading from go.mod")
+		tok := strings.Split(modPathWithoutVersion(upstream.Mod.Path), "/")
+		upstreamOrg = tok[len(tok)-2]
+
```

Comment:
The logic is much simpler 🎉! Code wise, I think this could be made simpler still.

In general, I think that we should avoid code like this:

```go
		var upstreamOrg string
		upstreamOrg = GetContext(ctx).UpstreamProviderOrg
		// break
		upstreamOrg = tok[len(tok)-2]
		return upstreamOrg
```

Instead, we should write:

```go
		upstreamOrg := GetContext(ctx).UpstreamProviderOrg
		// break
		// We expect tokens to be of the form:
		//
		//	github.com/${org}/${repo}/${path}
		//
		// The second chunk is the org name.
		return tok[len(tok)-2]
```

☝️ actually calls `tok[len(tok)-2]` into question. Most TF providers have a top level module (path = `github.com/${org}/${repo}`), but go allows nested modules (path = `github.com/${org}/${repo}/${nest-path}`). Since we are assuming GitHub, `tok[len(tok)-2]` will fail on nested paths, while `tok[1]` will always work.

```suggestion
		if org := GetContext(ctx).UpstreamProviderOrg; org != "" {
			stepv2.SetLabelf(ctx, "From Upgrade Config: %s", org)
			return org
		}
		// fall back to attempting to read from go.mod
		stepv2.SetLabel(ctx, "Fall back to reading from go.mod")
		tok := strings.Split(modPathWithoutVersion(upstream.Mod.Path), "/")
		return tok[1]

```

Reply:
that's a good catch!